paimon-common/src/main/java/org/apache/paimon/lookup/sort/BlockWriter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.lookup.sort;
+
+import org.apache.paimon.memory.MemorySlice;
+import org.apache.paimon.memory.MemorySliceOutput;
+import org.apache.paimon.utils.IntArrayList;
+
+import java.io.IOException;
+
+import static org.apache.paimon.lookup.sort.BlockAlignedType.ALIGNED;
+import static org.apache.paimon.lookup.sort.BlockAlignedType.UNALIGNED;
+
+/** Writer to build a Block. */
+public class BlockWriter {
+
```

Comment:
How about adding a header for Block that contains BlockType and onDiskSize. 
For BlockType, we can easily support multiple level block index like HFile. 
For onDiskSize, we can scan block data directly without reading it from block index for each block.

Reply:
Currently, there is already a `BlockAligndType` at the end of the block. But it is indeed extensible.

Reply:
Do you imply that we can extend `BlockAlignType` to `DATA_ALIGN`, `DATA_UNALIGN`, `INDEX`, etc.? Cool! How about write this `byte` in header of block with `onDiskSize`?

Reply:
> Do you imply that we can extend BlockAlignType to DATA_ALIGN, DATA_UNALIGN, INDEX, etc.?

Yes, but first we should figure out what is multiple level block index, and what is benifit.

> How about write this byte in header of block with onDiskSize?

I cannot understand good about `we can scan block data directly without reading it from block index for each block`.

> footer or header

I don't have any bias, they all seem the same as long as they don't affect the writing process